### PR TITLE
Remove more unused code and fix/add tests

### PR DESCRIPTION
**Pull request by ssjunnebo**
* Remove various bits of code that's no longer used
* Make existing tests run independent of system (e.g. without access to Charon etc)
* Minor bug fixes noticed while writing tests
* Add new tests
* Fix issue with uploading coverage reports to codecov (travis installed an older version of codecov for some reason. Force it to upgrade)

The tests for `CharonSession` can still only be run with access to charon-dev since I have not been able to figure out how to mock the request calls.

### CODE REVIEW

**Comment by ssjunnebo** [comment]
Test coverage is now at ~77%, [report here](https://codecov.io/gh/NationalGenomicsInfrastructure/ngi_pipeline/tree/98138fb3b613ffc4380fa771303ae2ea6f872e14).

**Review by matrulda — APPROVED** [review]
Wow, awesome work! 🎉 I think it looks good, left a few minor comments. 

Let me know if you want me to merge or want to get some more reviews.

File: ngi_pipeline/tests/database/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def test_load_charon_variables_w_conf():
+    conf = {'charon': {'charon_api_token': 'token', 
+            'charon_base_url': 'URL'}}
+    expected_charon_variables = {'charon_api_token': 'token', 
+                                    'charon_base_url': 'URL'}
+    got_charon_variables = load_charon_variables(conf)
+    assert got_charon_variables == expected_charon_variables
+
+def test_load_charon_variables_wo_conf():
+    # Set CHARON_BASE_URL="charon-url" and CHARON_API_TOKEN="token" for this test to work (see .travis.yml)
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/tests/database/test_utils.py:12]: Neat! Had no idea about this 😄 Fixed 👍

File: ngi_pipeline/tests/engines/piper_ngi/test_parsers.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                                    '10 100 50']
+        with open(self.genome_results_file, 'w') as f:
+            f.write("\n".join(genome_results_content))
+
+    @classmethod
+    def tearDownClass(self):
+        shutil.rmtree(self.tmp_dir)
+
+    @mock.patch('ngi_pipeline.engines.piper_ngi.parsers.parse_qualimap_reads')
+    def test_parse_results_for_workflow(self, mock_parse):
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/tests/engines/piper_ngi/test_parsers.py:27]: I don't think it's used anywhere, but I've tried to keep all the piper stuff as it is. I guess we can remove the whole engine eventually instead?

File: ngi_pipeline/tests/engines/piper_ngi/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+        expected_libpreps = {'A': ['B']}
+
+        self.assertEqual(got_libpreps, expected_libpreps)
+
+    @mock.patch('ngi_pipeline.engines.piper_ngi.utils.CharonSession')
+    def test_get_valid_seqruns_for_sample(self, mock_charon):
+        mock_charon().sample_get_libpreps.return_value = {'libpreps': [{'qc': 'PASS',
+                                                                        'libprepid': 'A'}]}
+        mock_charon().libprep_get_seqruns.return_value = {'seqruns': [{'seqrunid': 'B'}]}
+        mock_charon().seqrun_get.return_value = {}
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/tests/engines/piper_ngi/test_utils.py:106]: Yup, you're right 😄 I'll remove it 🙂

File: ngi_pipeline/tests/engines/piper_ngi/test_workflows.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+        with self.assertRaises(NotImplementedError):
+            missing_workflow_name = 'not_implemented'
+            got_command_line = workflows.return_cl_for_workflow(missing_workflow_name, qscripts_dir_path, setup_xml_path)
+
+    def test_workflow_merge_process_variantcall(self):
+        qscripts_dir_path = 'path'
+        setup_xml_path = 'other/path'
+        config = {'slurm': {'time': '1-00:00:00'},
+                    'piper': {'threads': 2,
```
[the remark this answers is not in the document]
Comment (ssjunnebo) [reply ngi_pipeline/tests/engines/piper_ngi/test_workflows.py:44]: Agreed! I think there are quite a few formatting issues like this 😬  I'll go through the rest as well and fix it 🙂

**Comment by ssjunnebo** [comment]
> Wow, awesome work! 🎉 I think it looks good, left a few minor comments.
> 
> Let me know if you want me to merge or want to get some more reviews.

I think I've adressed everything now, and I also went through most of the tests again to fix the formatting. Not sure if anyone else want's to have a look as well before merging? 🙂

File: ngi_pipeline/engines/piper_ngi/launchers.py [hunk 5]
```diff
@@ -41,9 +41,7 @@
                                           match_files_under_dir
 from ngi_pipeline.utils.classes import with_ngi_config
 from ngi_pipeline.utils.filesystem import fastq_files_under_dir, is_index_file
-from ngi_pipeline.utils.parsers import parse_lane_from_filename, \
-                                       find_fastq_read_pairs_from_dir, \
-                                       get_flowcell_id_from_dirtree
+#from ngi_pipeline.utils.parsers import get_flowcell_id_from_dirtree
```
Comment (aanil) [inline ngi_pipeline/engines/piper_ngi/launchers.py:44]: Are we going to something with this in the future?

**Review by aanil — APPROVED** [review]
👍  Looks good!

File: ngi_pipeline/engines/piper_ngi/launchers.py (hunk 5 above)
Comment (ssjunnebo) [reply ngi_pipeline/engines/piper_ngi/launchers.py:44]: Ah, no, I removed the function but missed this line. I'll remove it now, well spotted! 😄

File: ngi_pipeline/tests/engines/piper_ngi/test_parsers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (matrulda) [reply ngi_pipeline/tests/engines/piper_ngi/test_parsers.py:27]: Yeah, makes sense 👍

**Comment by matrulda** [comment]
Nice, looks good 👍  Do you want me to merge?

**Comment by ssjunnebo** [comment]
> Nice, looks good 👍 Do you want me to merge?

Sounds good! 😄

**Pull request merged**